units/en/unit3/hands-on.mdx

```diff
@@ -0,0 +1,320 @@
+
+⬇️ Here is an example of what **you will achieve** ⬇️
+
+```python
+%%html
+<video controls autoplay><source src="https://huggingface.co/ThomasSimonini/ppo-SpaceInvadersNoFrameskip-v4/resolve/main/replay.mp4" type="video/mp4"></video>
+```
+
+### 🎮 Environments:
+
+- SpacesInvadersNoFrameskip-v4
+
+### 📚 RL-Library:
+
+- [RL-Baselines3-Zoo](https://github.com/DLR-RM/rl-baselines3-zoo)
+
+## Objectives 🏆
+
+At the end of the notebook, you will:
+
+- Be able to understand deeper **how RL Baselines3 Zoo works**.
+- Be able to **push your trained agent and the code to the Hub** with a nice video replay and an evaluation score 🔥.
+
+
+## Prerequisites 🏗️
+Before diving into the notebook, you need to:
+
+🔲 📚 **[Study Deep Q-Learning by reading Unit 3](https://huggingface.co/deep-rl-course/unit3/introduction)**  🤗
+
+We're constantly trying to improve our tutorials, so **if you find some issues in this notebook**, please [open an issue on the Github Repo](https://github.com/huggingface/deep-rl-class/issues).
+
+# Let's train a Deep Q-Learning agent playing Atari' Space Invaders 👾 and upload it to the Hub.
+
+To validate this hands-on for the certification process, you need to push your trained model to the Hub and **get a result of >= 500**.
+
+To find your result, go to the leaderboard and find your model, **the result = mean_reward - std of reward**
+
+For more information about the certification process, check this section 👉 https://huggingface.co/deep-rl-course/en/unit0/introduction#certification-process
+
+## Set the GPU 💪
+- To **accelerate the agent's training, we'll use a GPU**. To do that, go to `Runtime > Change Runtime type`
+
+<img src="https://huggingface.co/datasets/huggingface-deep-rl-course/course-images/resolve/main/en/notebooks/gpu-step1.jpg" alt="GPU Step 1">
+
+- `Hardware Accelerator > GPU`
+
+<img src="https://huggingface.co/datasets/huggingface-deep-rl-course/course-images/resolve/main/en/notebooks/gpu-step2.jpg" alt="GPU Step 2">
+
+## Create a virtual display 🔽
+
+During the notebook, we'll need to generate a replay video. To do so, with colab, **we need to have a virtual screen to be able to render the environment** (and thus record the frames).
+
+Hence the following cell will install the librairies and create and run a virtual screen 🖥
+
+```bash
+apt install python-opengl
+apt install ffmpeg
+apt install xvfb
+pip3 install pyvirtualdisplay
+```
+
+```bash
+apt-get install swig cmake freeglut3-dev
+```
+
+```bash
+pip install pyglet==1.5.1
+```
+
+```python
+# Virtual display
+from pyvirtualdisplay import Display
+
+virtual_display = Display(visible=0, size=(1400, 900))
+virtual_display.start()
+```
+
+## Clone RL-Baselines3 Zoo Repo 📚
+You can now directly install from python package `pip install rl_zoo3` but since we want **the full installation with extra environments and dependencies** we're going to clone `RL-Baselines3-Zoo` repository and install from source.
```

Comment:
```suggestion
You could directly install from the Python package (`pip install rl_zoo3`), but since we want **the full installation with extra environments and dependencies**, we're going to clone the `RL-Baselines3-Zoo` repository and install from source.
```